packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2613,26 +3061,32 @@ export class TransactionController extends BaseControllerV1<
     this.hub.emit('transaction-status-update', { transactionMeta });
   }
 
-  private getNonceTrackerPendingTransactions(address: string) {
+  private getNonceTrackerPendingTransactions(
+    chainId: string | undefined,
+    address: string,
+  ) {
     const standardPendingTransactions = this.getNonceTrackerTransactions(
       TransactionStatus.submitted,
       address,
+      chainId,
     );
 
-    const externalPendingTransactions =
-      this.getExternalPendingTransactions(address);
+    // TODO(JL): modify getExternalPendingTransactions in extension to accept a chainId to filter for smartTransactions by chainId
```

Comment:
note this somewhere else then delete this comment

Reply:
working on this

Reply:
https://github.com/MetaMask/MetaMask-planning/issues/1019

Reply:
https://github.com/MetaMask/core/pull/3643/commits/b6b4e3769a9a3dfbd7a5c6ae68c0e4c3b7334fd5